coap/api/transport.go

```diff
@@ -68,7 +69,9 @@ func MakeCOAPHandler(svc coap.Service, tc mainflux.ThingsServiceClient, l log.Lo
 	pingPeriod = pp
 	r := mux.NewRouter()
 	r.Handle("/channels/{id}/messages", gocoap.FuncHandler(receive(svc))).Methods(gocoap.POST)
+	r.Handle("/channels/{id}/messages/{subtopic:[/a-zA-Z0-9-_*>+#]+}", gocoap.FuncHandler(receive(svc))).Methods(gocoap.POST)
```

Comment:
I still do not see why you cannot have subtopic name like `temp!%%%(99)/room?$&/~33`

I.e. I think you should enable all special chars allowed by MQTT protocol. For this you must open MQTT spec and take a look at topic rules: http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc398718106

```
4.7.3 Topic semantic and usage
The following rules apply to Topic Names and Topic Filters:

All Topic Names and Topic Filters MUST be at least one character long [MQTT-4.7.3-1]
Topic Names and Topic Filters are case sensitive
Topic Names and Topic Filters can include the space character
A leading or trailing ‘/’ creates a distinct Topic Name or Topic Filter
A Topic Name or Topic Filter consisting only of the ‘/’ character is valid 
Topic Names and Topic Filters MUST NOT include the null character (Unicode U+0000) [Unicode] [MQTT-4.7.3-2]
Topic Names and Topic Filters are UTF-8 encoded strings, they MUST NOT encode to more than 65535 bytes [MQTT-4.7.3-3]. See Section 1.5.3
```

When I read this: `Topic Names and Topic Filters are UTF-8 encoded strings` - this basically means any possible [UTF-8 character](https://en.wikipedia.org/wiki/List_of_Unicode_characters)